### PR TITLE
Add "reporter" and "assignee" fields in subscription filters

### PR DESCRIPTION
**Pull request by shigalovalexs**
**Summary**:
Added "reporter" and "assignee" fields to subscriptions modal to filter notifications in channels by users
Also added a field "reporter" to the issue create dialog if the project supports it
image
image

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-plugin-jira/issues/891

### CODE REVIEW

File: .github/workflows/build-plugin-jira.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Stuff to do CI for mattermost plugins
```
Comment (mickmister) [inline .github/workflows/build-plugin-jira.yml:1]: @shigalovalexs I'm thinking this is out of scope of this PR. We're currently using CircleCI for the plugin projects. Also, if we were to add GitHub actions, we would want to support it holistically across projects rather than adding it to individual repos like this.

**Review by mickmister — COMMENTED** [review]
Thanks for this @shigalovalexs! LGTM :+1: just one suggestion to remove the GitHub actions added in this PR. cc @hanzei for discussion on this

File: webapp/src/utils/jira_issue_metadata.tsx [hunk 2]
```diff
@@ -264,6 +276,10 @@ export function isLabelField(field: JiraField | FilterField): boolean {
     return field.schema.system === 'labels' || field.schema.custom === 'com.atlassian.jira.plugin.system.customfieldtypes:labels';
 }
 
+export function isUserField(field: JiraField | FilterField): boolean {
+    return field.schema.type === 'user' || field.schema.custom === 'com.atlassian.jira.plugin.system.customfieldtypes:userpicker';
```
Comment (javaguirre) [inline webapp/src/utils/jira_issue_metadata.tsx:280]: Could the string 'com.atlassian.....' be a constant more to the top of the file? It seems some kind of "magic string".

File: webapp/src/utils/jira_issue_metadata.tsx [hunk 3]
```diff
@@ -179,7 +179,7 @@ function isValidFieldForFilter(field: JiraField): boolean {
     }
 
     return allowedTypes.includes(type) || (custom && acceptedCustomTypesForFilters.includes(custom)) ||
```
Comment (javaguirre) [inline webapp/src/utils/jira_issue_metadata.tsx:181]: I find this part a bit difficult to follow and understand, but this would be a future refactor.

**Review by javaguirre — CHANGES_REQUESTED** [review]

File: webapp/src/utils/jira_issue_metadata.tsx (hunk 3 above)
Comment (mickmister) [reply webapp/src/utils/jira_issue_metadata.tsx:181]: @javaguirre Yeah I too think `option` and `user` should be added to `allowedTypes`. I don't see any logic that would bring if we did that

**Comment by hanzei** [comment]
@shigalovalexs Is the target branch here really correct?

**Comment by hanzei** [comment]
@shigalovalexs Are you open to signing the CLA?

**Comment by shigalovalexs** [comment]
@hanzei Hi, yes, I`m agree to sign the CLA

**Comment by hanzei** [comment]
@shigalovalexs Could you please sign the form at https://mattermost.com/mattermost-contributor-agreement/?

File: plugin.json [hunk 4]
```diff
@@ -4,9 +4,9 @@
     "description": "Atlassian Jira plugin for Mattermost.",
     "homepage_url": "https://github.com/mattermost/mattermost-plugin-jira",
     "support_url": "https://github.com/mattermost/mattermost-plugin-jira/issues",
-    "release_notes_url": "https://github.com/mattermost/mattermost-plugin-jira/releases/tag/v3.2.0",
+    "release_notes_url": "https://github.com/mattermost/mattermost-plugin-jira/releases/tag/v3.2.1",
     "icon_path": "assets/icon.svg",
-    "version": "3.2.0",
+    "version": "3.2.1",
```
Comment (hanzei) [inline plugin.json:9]: Unrelated change?

File: server/manifest.go [hunk 5]
```diff
@@ -7,5 +7,5 @@ var manifest = struct {
 	Version string
 }{
 	ID:      "jira",
-	Version: "3.2.0",
+	Version: "3.2.1",
```
Comment (hanzei) [inline server/manifest.go:10]: Same

File: plugin.json (hunk 4 above)
Comment (hanzei) [reply plugin.json:9]: This is not resolved
